Source/Core/Core/IOS/USB/Emulated/Infinity.cpp

```diff
@@ -0,0 +1,907 @@
+  }
+  std::array<u8, 16> figure_data = GenerateBlankFigureData(figure_num);
+
+  if (figure_data[1] == 0x00)
+    return false;
+
+  std::array<u8, 16> infinity_aes_key = GenerateInfinityFigureKey(sha1_calc);
+
+  // Create AES Encrypt context based on AES key, use this to encrypt the character data and 4 blank
+  // blocks
+  std::unique_ptr<Common::AES::Context> ctx =
+      Common::AES::CreateContextEncrypt(infinity_aes_key.data());
+  std::array<u8, 16> encrypted_block = {};
+  std::array<u8, 16> encrypted_blank = {};
+
+  ctx->CryptIvZero(figure_data.data(), encrypted_block.data(), figure_data.size());
+  ctx->CryptIvZero(BLANK_BLOCK.data(), encrypted_blank.data(), BLANK_BLOCK.size());
+
+  // Copy encrypted data and UID data to the Figure File
+  memcpy(&file_data[0], uid_data.data(), uid_data.size());
+  memcpy(&file_data[16], encrypted_block.data(), encrypted_block.size());
+  memcpy(&file_data[16 * 0x04], encrypted_blank.data(), encrypted_blank.size());
+  memcpy(&file_data[16 * 0x08], encrypted_blank.data(), encrypted_blank.size());
+  memcpy(&file_data[16 * 0x0C], encrypted_blank.data(), encrypted_blank.size());
+  memcpy(&file_data[16 * 0x0D], encrypted_blank.data(), encrypted_blank.size());
+
+  DEBUG_LOG_FMT(IOS_USB, "File Data: \n{}", HexDump(file_data.data(), file_data.size()));
+
+  inf_file.WriteBytes(file_data.data(), file_data.size());
+
+  return true;
+}
+
+const std::array<std::pair<const char*, const u32>, 104>& InfinityBase::GetFigureList()
+{
+  return list_infinity_figures;
+}
+
+std::string InfinityBase::FindFigure(u32 number) const
+{
+  for (auto it = list_infinity_figures.begin(); it != list_infinity_figures.end(); it++)
+  {
+    if (it->second == number)
+    {
+      return it->first;
+    }
+  }
+  return "";
+}
+
+u8 InfinityBase::DeriveFigurePosition(u8 position)
+{
+  // In the added/removed response, position needs to be 1 for the hexagon, 2 for Player 1 and
+  // Player 1's abilities, and 3 for Player 2 and Player 2's abilities. Abilities are in positions
+  // > 2 in the UI (3/5 for player 1, 4/6 for player 2) so decrement the position until < 2.
+
+  while (position > 2)
+    position -= 2;
+
+  position++;
+  return position;
+}
+
+InfinityFigure& InfinityBase::GetFigureByOrder(u8 order_added)
+{
+  for (u8 i = 0; i < m_figures.size(); i++)
+  {
+    if (m_figures[i].order_added == order_added)
+    {
+      return m_figures[i];
+    }
+  }
+  // This should never reach this statement, but return 0 reference to supress warnings
+  ASSERT_MSG(IOS_USB, false, "Unable to find Disney Figure requested by game");
+  return m_figures[0];
+}
+
+std::array<u8, 16> InfinityBase::GenerateInfinityFigureKey(const std::vector<u8>& sha1_data)
+{
+  Common::SHA1::Digest digest = Common::SHA1::CalculateDigest(sha1_data);
+  // Infinity AES keys are the first 16 bytes of the SHA1 Digest, every set of 4 bytes need to be
+  // reversed due to endianness
+  std::array<u8, 16> infinity_aes_key = {};
```

Comment:
Just "key" is enough - we're in a file called Infinity.cpp, in a member function of a class called InfinityBase, and the function name has Infinity in it. I think it's clear what kind of key this is :)